server/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java

```diff
@@ -250,8 +249,11 @@ protected void parseCreateField(ParseContext context) throws IOException {
         if (value == null) {
             return;
         }
-        if (fieldType().isSearchable() || fieldType.stored()) {
-            context.doc().add(new Field(fieldType().name(), value ? "T" : "F", fieldType));
+        if (fieldType().isSearchable()) {
+            context.doc().add(new Field(fieldType().name(), value ? "T" : "F", Defaults.FIELD_TYPE));
+        }
+        if (stored) {
+            context.doc().add(new StoredField(fieldType().name(), value ? "T" : "F"));
```

Comment:
seems like if the field is searchable and stored too, we end up adding two fields, while before we would only add one?

Reply:
Adding a field with a `FieldType` that has `stored` set to `true` and adding a separate `StoredField` amount to the same thing.  This way, we don't need to change the field type at all, and it's clearer what's happening at the point where we're adding documents.

Reply:
It was odd that we were using `Field` to get stored fields but doc values. `Field` just does a bunch of stuff.